pkg/codegen/hcl2/model/binder_expression.go

```diff
@@ -769,6 +769,22 @@ func (b *expressionBinder) bindUnaryOpExpression(syntax *hclsyntax.UnaryOpExpr)
 	if tokens == nil {
 		tokens = _syntax.NewUnaryOpTokens(syntax.Op)
 	}
+
+	// simplify UnaryOperation(Negate, Number(N)) into Number(-N)
+	if syntax.Op == hclsyntax.OpNegate {
+		if lit, ok := operand.(*LiteralValueExpression); ok && lit.Value.Type().Equals(cty.Number) {
+			negated := &LiteralValueExpression{
+				Value:  lit.Value.Negate(),
+				Syntax: lit.Syntax,
```

Comment:
Your losing the syntax and tokens of the operator here, and Syntax.Val will be the wrong value. Again I think you can just copy what I did in https://github.com/pulumi/pulumi/pull/16939/files#diff-0925bf1716dbbe0ef4c7363db8d274d28213f80d7620a1a6a6213cd775c94bf5R786

Reply:
> Your losing the syntax and tokens of the operator here

Nice catch!

Reply:
Done ✅